### PR TITLE
Add options view

### PR DESCRIPTION
**Pull request by arvidsandin**

### CODE REVIEW

File: src/components/ask-manager/ask-manager.tsx [hunk 1]
```diff
@@ -28,31 +54,65 @@ export class AskManager {
...
     this.acceptCategories(this.categories);
   };
+  acceptSelectedCookies = () => {
+    let selectedCategories = [];
+    for (const option of this.categoryCheckboxes) {
+      if (option.checked) {
+        selectedCategories.push(option.value);
+      }
+    }
+    this.acceptCategories(selectedCategories);
```
Comment (R0tenur) [inline src/components/ask-manager/ask-manager.tsx:80]: Alternative how to write this a bit more compact:
```ts
selectedCategories.filter(o => o.checked).map(o => o.value);
```

This is probably not optimal for the sake of performance, but you will not have that many categories.

If performance is more important:
```ts
selectedCategories.reduce((filtered, option) => {
  if (option.checked) {
     filtered.push(option.value);
  }
  return filtered;
}, []);
```
But then the readability you have now is better

File: src/components/ask-manager/ask-manager.tsx [hunk 2]
```diff
@@ -28,31 +54,65 @@ export class AskManager {
...
-        <div class="consent-box">
-          <p class="info-text">
-            This website uses cookies for functional, analytical and marketing purposes. Read more in our <a href="https://example.com">privacy policy</a> You can manage your
-            choices at any time.
-          </p>
-          <button onClick={this.showOptions}>More options</button>
-          <button onClick={this.acceptAllCookies}>Reject non-essential</button>
-          <button onClick={this.rejectAllCookies}>Accept all</button>
-        </div>
+        {this.isInOptionsView ? (
```
Comment (R0tenur) [inline src/components/ask-manager/ask-manager.tsx:87]: Split into seperate components. 

If not using a framework for selecting what component to show I would consider an object in the component:

```tsx
views = {
  'fullscreen-options': <FullscreenOptions/>, // Pointing to other components
  'bottom-options': <BottomOptions/>,
  etc
}
type View = keyof views;

selectedView: View;

```

Then you could do some conditional rendering by
```html
<div class="dimmable-backdrop">
    {views[selectedView]}
</div>
```

**Review by R0tenur — CHANGES_REQUESTED** [review]
Generally I would like you to look into the access modifiers on the functions.

public/private etc

**Comment by arvidsandin** [comment]
> Generally I would like you to look into the access modifiers on the functions.
> 
> public/private etc

If I make the functions private (which most, if not all should be), they can no longer be executed in the unit tests. I still haven't figured out how to handle that better

**Comment by R0tenur** [comment]
Generally it's only the public api that should be tested.
The private stuff can be tester based on the  side effects and outcome

File: src/components/ask-manager/ask-manager.tsx (hunk 2 above)
Comment (arvidsandin) [reply src/components/ask-manager/ask-manager.tsx:87]: Will be fixed in another pull request

File: src/components/ask-manager/ask-manager.tsx [hunk 3]
```diff
@@ -9,50 +9,105 @@ export class AskManager {
...
    * can be any string that can be read by Date()
    */
-  @Prop() cookiePolicyLastUpdated!: string;
+  private cookiePolicyLastUpdated: string = null;
 
-  cookieConsent = JSON.parse(localStorage.getItem('cookie-consent')) || {
+  /**
+   * key to use when storing the consent in localStorage
+   */
+  @Prop() storageName: string = 'cookie-consent';
```
Comment (R0tenur) [inline src/components/ask-manager/ask-manager.tsx]: Would say everything should be part of options

**Review by R0tenur — CHANGES_REQUESTED** [review]

**Review by R0tenur — APPROVED** [review]

File: src/components/ask-manager/ask-manager.tsx (hunk 2 above)
Comment (arvidsandin) [reply src/components/ask-manager/ask-manager.tsx:87]: Solved in #5

**Pull request merged**